modules/consensus/synchronize.go

```diff
@@ -43,6 +44,9 @@ func (s *ConsensusSet) receiveBlocks(conn modules.PeerConn) error {
 			// Blocks received during synchronize aren't trusted; activate full
 			// verification.
 			lockID := s.mu.Lock()
+			if !s.db.open {
+				return errors.New("database not open")
```

Comment:
It happens when you close the database and there are still threads running. As the closing happens after the API shuts down, I believe there are only two spots where this guard needs to happen: here and in the synchronize loop.

I'm not accounting for random occurrences like the disk being disconnected; that will just panic.